ql/ql/src/queries/style/OmittableExists.ql

```diff
@@ -0,0 +1,21 @@
+/**
+ * @name Omittable 'exists' argument
+ * @description Writing 'exists(x | pred(x))' is bad practice and can be omitted.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id ql/omittable-exists
+ * @tags maintainability
+ */
+
+import ql
+
+from VarDecl existsArgument, VarAccess use
+where
+  existsArgument = any(Exists e).getAnArgument() and
+  use = unique( | | existsArgument.getAnAccess()) and
+  exists(Call c, int argPos | c.getArgument(argPos) = use |
+    not existsArgument.getType().getASuperType+() = c.getTarget().getParameterType(argPos)
```

Comment:
I would rather that the logic was flipped. 
Such that you confirm the class hierarchy is "good", rather than showing a case where it's "bad".  

I think that makes it more readable.  

```suggestion
   existsArgument.getType() = c.getTarget().getParameterType(argPos).getASuperType*()
```